### DISCUSSION TITLE
Some Bugs

### DISCUSSION THREAD

**Discussion by py4xx** [mbridak/not1mm#242] [General]
Hello...
I worked on RAC Winter this weekend and noticed some bugs.
- During the contest, Not1mm would suddenly close, and when it came back, it would come back without the macros configuration.
![Captura de tela de 2024-12-27 23-15-19](https://github.com/user-attachments/assets/f71267bc-7ece-4898-a6ff-9bbe067d5c58)

- I still miss CQ AUTO.
- Several times, when calling CQ, when placing a call in the call field, CQ would not interrupt, having to manually cancel the CQ.
- Is there a way to force the saving of the contact, without having to click on the "Log it" button? Something like at the end of the "TU" macro like "TU <savelog> "? I ask because since RAC Winter is not yet in the Contests list, I used REF, but exch is different, and it did not log when pressing ENTER. So I had to keep clicking with the mouse at the end of the contact to save...
And finally, a suggestion:
Is there a way to make the user himself include the contest he wants to compete in? There are many contests and it is really difficult for just one person to put all the contests on not1mm... This way, if I want to do a contest that is not yet on not1mm, I could include it myself...
We are on the right track...
Forte 73.

**Comment by mbridak** [comment]
Hello, thanks for your report.

- **Random closing**: If the program randomly closes it has crashed for some reason. It would help if you launch not1mm from a terminal. This won't make it not crash, but it will show you why and where in the program it had crashed. You can then paste the traceback that gets spit out when it crashed so we can fix it.
- **Missing Auto CQ:** I do to. I operated the ARRL 160 and became weary pressing F1 into the void.
- **CQ not interrupting when typing in the callsign field:** This is realted to the previous auto cq point.
- **Force saving contact:** I'm on board with this. I should make a `{LOGIT}` macro. This wasn't a problem before ESM, but now it is.
- **Using REF for RAC:** You would have probably had a better experience if you used the RAC Canada Day contest plugin for RAC Winter. As far as I could tell they have the exact exchange and scoring.
- **User defined contests:** I do not believe I'm clever enough to pull this off. So for someone to be able to add a contest, they would need to learn Python, maybe some SQL and the general way the program works (which is with bubblegum and a prayer) roll up their sleeves and get to coding.

Cheers.

**Reply by py4xx** [reply]
Random closing: If the program randomly closes it has crashed for some reason. It would help if you launch not1mm from a terminal. This won't make it not crash, but it will show you why and where in the program it had crashed. You can then paste the traceback that gets spit out when it crashed so we can fix it.
_As it's random, I'll try via the terminal to see if it happens again._

    Missing Auto CQ: I do to. I operated the ARRL 160 and became weary pressing F1 into the void.
   _KKKK It's really tiring..._

    CQ not interrupting when typing in the callsign field: This is realted to the previous auto cq point.
    
    _Roger, Roger.._

Force saving contact: I'm on board with this. I should make a {LOGIT} macro. This wasn't a problem before ESM, but now it is.
  _TNX_
  
    Using REF for RAC: You would have probably had a better experience if you used the RAC Canada Day contest plugin for RAC Winter. As far as I could tell they have the exact exchange and scoring.
  _My fault... I should have noticed this... I'll pay more attention next time._

    User defined contests: I do not believe I'm clever enough to pull this off. So for someone to be able to add a contest, they would need to learn Python, maybe some SQL and the general way the program works (which is with bubblegum and a prayer) roll up their sleeves and get to coding.

_Wow... I don't know anything, nothing about python or SQL KKK... I imagined that not1mm could have this facility like DXLog, but it must be a different system._

_ I still haven't figured out the macros... Where is the macro file? Because I have to configure it every time I open a contest._

**Reply by mbridak** [reply]
- **Where are the macro files:** It depends. [See here](https://github.com/mbridak/not1mm?tab=readme-ov-file#various-data-file-locations).
- But a good guess is `~/.local/share/not1mm/`
- There are multiple macro files. One for each basic mode. cwmacros.txt for CW, ssbmacros.txt for phone and rttymacros.txt for data. Once you select a new contest a folder is created with the contests name. A copy of the base macros is copied to the contests folder. This way you can maintain different macros for each contest.
- You can edit the current contests mode macro file by going to `FILE -> Edit Macros` once saved toggle between Run and S&P to force it to reload. It will now be available to reload when the program is re-run.

**Comment by mbridak** [comment]
Added `{LOGIT}` macro.
You can also press `CTRL =` to log it.

**Reply by py4xx** [reply]
TNX ... 
It will be of great value.
